### ISSUE TITLE
CASE statement around Field Type

### ISSUE THREAD

**Issue by davidwCP** [Microsoft/AL#1365]
Version: NAV2018 - 11.0.19394.0

I was trying to recreate a modification made in 2017 in the new AL environment, but hit the following. I've isolated the issue into a smaller subset of code around case statements and comparing the field type value. The confirms are just to help break out the loop but its noticeable on the first field.

The code is the same in both AL and 2018:
 - FieldRec : Record : Field
```
FieldRec.SETRANGE(TableNo,23);
FieldRec.SETRANGE(Class,FieldRec.Class::Normal);
IF FieldRec.FINDSET THEN
  REPEAT
    CASE FieldRec.Type OF
      FieldRec.Type::Text,FieldRec.Type::Code: 
        IF NOT CONFIRM('Field Type: %1 - Type Found',TRUE,FORMAT(FieldRec.Type)) THEN
          ERROR('');
      ELSE 
        IF NOT CONFIRM('Field Type: %1 - Type Not Found',TRUE,FORMAT(FieldRec.Type)) THEN
          ERROR('');
    END;
  UNTIL FieldRec.NEXT = 0
```
The first field in the Vendor table is a code so when running from the dev environment we recieve:
![image](https://user-images.githubusercontent.com/35226476/34674371-9274b1d2-f47d-11e7-98cd-b49e8b379e7f.png)

If we run that in AL which opens on the local 2018 instance, every loop gives the same as below:
![image](https://user-images.githubusercontent.com/35226476/34674544-45bbeb02-f47e-11e7-8e95-751bca463b89.png)

I've tried encasing the Text and Code on the options in speech marks, as I can see that on VS Code they are highlighted as though there variables. This made no difference however:
![image](https://user-images.githubusercontent.com/35226476/34674859-6785adb2-f47f-11e7-9be5-752cbe68a297.png)

The only way I've got the first confirm to fire is writing the value of FieldRec.Type into a text variable first, and then performing the case statement on that instead.

**Comment by anzwdev** [comment]
I've just checked it, for "Code" fields, Field.Type contains 31489 value in both C/AL in AL, but Field.Type::Code const value in AL has incorrect value of 10. It is v1 blocker for us as we have plenty of code checking field types everywhere in our extension.

**Comment by NicolaiEsbensen** [comment]
Also a big issue for us.

**Comment by Pieterheinevdm** [comment]
in which nav2018 cu version is this going to be fixed?  in 11.0.23019 the problem still occurs.
